types/jquery/jquery-tests.ts

```diff
@@ -4342,24 +4446,366 @@ function JQueryStatic() {
     }
 
     function when() {
-        const t = $.ajax() as JQuery.jqXHR<string>;
-        const u = $.ajax() as JQuery.jqXHR<number>;
-        const v = $.ajax() as JQuery.jqXHR<boolean>;
+        function Promise3() {
+            const t = $.ajax() as JQuery.jqXHR<string>;
+            const u = $.ajax() as JQuery.jqXHR<number>;
+            const v = $.ajax() as JQuery.jqXHR<boolean>;
+
+            // 3 parameters
+            {
+                const w = $.when(t, u, v);
+
+                w.then((a, b, c) => {
```

Comment:
Actually, it looks like the spec doesn't preclude more than 1 argument (https://promisesaplus.com/#point-26).

> If onFulfilled is a function:
> - it must be called after promise is fulfilled, with promise’s value as its first argument.

> If onRejected is a function,
> - it must be called after promise is rejected, with promise’s reason as its first argument.